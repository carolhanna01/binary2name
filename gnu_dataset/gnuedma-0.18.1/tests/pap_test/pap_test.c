/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdlib.h>

#include <edma.h>

main(int argc,char *argv[])
{
  ESint32          res;
  OBJID            id;
  OBJID            id_sc,id_sc1;

  /* Initialize EDMA System*/
  EDMAInit();

  /* Put your C code here */
  edma_printf ("\n Preferer Anchor Point Test");
  edma_printf ("+ Creating NFILE object...");
  if ((id = edma_new_obj ("NFILE"))==-1)  {
    edma_printf ("Can't create SUBCLASE2 object");
    EDMAEnd();
    exit(1);
  }
  edma_printf ("Object Report for NFILE object");
  edma_obj_report (id);
  edma_printf ("--------------------------------");

  edma_printf ("+ Upcasting object NFILE to STREAM...");
  id_sc1 = edma_upcast_obj (id, "STREAM");
  edma_printf ("Object Report for STREAM superobject");
  edma_obj_report (id_sc1);
  edma_show_subobjects_down (id_sc1, "****", 0);
  edma_printf ("--------------------------------------");

  edma_free_obj (id);

  /* Shutdown EDMA Suystem */
  EDMAEnd();
  return 0;
}
