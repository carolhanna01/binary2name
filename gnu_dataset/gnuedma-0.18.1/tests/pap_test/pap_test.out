
GNU/EDMA 0.16.4 build [Jan 30 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK

 Preferer Anchor Point Test
+ Creating NFILE object...
Object Report for NFILE object
==OBJECT REPORT==================================================
Object (0) of class (NFILE)
Top Level Object. Object haven't father
No PseudoFather information
Object has 1 superobjects
    Id : (STREAM) Class : (STREAM) Class attached Object: (STREAM)
Object has 0 subobjects
Virtual Methods in object
    Nobody overrides Method [open]
    Nobody overrides Method [close]
    Nobody overrides Method [read]
    Nobody overrides Method [write]
    Nobody overrides Method [puts]
    Nobody overrides Method [putc]
    Nobody overrides Method [gets]
    Nobody overrides Method [getc]
    Nobody overrides Method [flush]
    Nobody overrides Method [seek]
    Nobody overrides Method [tell]
    Nobody overrides Method [rewind]
    Nobody overrides Method [eof]
    Nobody overrides Method [error]
    Nobody overrides Method [Load]
    Nobody overrides Method [Save]
==END OBJECT REPORT==========================================
--------------------------------
+ Upcasting object NFILE to STREAM...
Object Report for STREAM superobject
==OBJECT REPORT==================================================
Object (1) of class (STREAM)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 1 subobjects
    Id : (IMP) Class : (NFILE) Class attached Object: (NFILE)
Virtual Methods in object
    Nobody overrides Method [open]
    Nobody overrides Method [close]
    Nobody overrides Method [read]
    Nobody overrides Method [write]
    Nobody overrides Method [puts]
    Nobody overrides Method [putc]
    Nobody overrides Method [gets]
    Nobody overrides Method [getc]
    Nobody overrides Method [flush]
    Nobody overrides Method [seek]
    Nobody overrides Method [tell]
    Nobody overrides Method [rewind]
    Nobody overrides Method [eof]
    Nobody overrides Method [error]
    Nobody overrides Method [Load]
    Nobody overrides Method [Save]
==END OBJECT REPORT==========================================
  IMP ==> NFILE [0]
**** ==> STREAM [1]
--------------------------------------
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
